perf/intkey_workload/src/intkey_transformer.rs

```diff
@@ -0,0 +1,150 @@
+/// an IntKeyPayload into a Sawtooth Transaction.
+pub struct IntKeyTransformer<'a> {
+    unsatisfiable: f32,
+    wildcard: f32,
+    num_names: usize,
+    signer: &'a signing::Signer<'a>,
+
+    addresser: IntKeyAddresser,
+
+    rng: StdRng,
+
+    txn_id_by_name: HashMap<String, String>,
+    txn_id_and_name: Vec<(String, String)>,
+}
+
+impl<'a> IntKeyTransformer<'a> {
+    pub fn new(
+        signer: &'a signing::Signer<'a>,
+        seed: &[usize],
+        unsatisfiable: f32,
+        wildcard: f32,
+        num_names: usize,
+    ) -> IntKeyTransformer<'a> {
+        IntKeyTransformer {
+            unsatisfiable: unsatisfiable,
+            wildcard: wildcard,
+            num_names: num_names,
+            signer: signer,
+            rng: SeedableRng::from_seed(seed),
+            addresser: IntKeyAddresser::new(),
+            txn_id_by_name: HashMap::new(),
+            txn_id_and_name: Vec::new(),
+        }
+    }
+
+    fn payload_to_cbor_bytes(&mut self, payload: &IntKeyPayload) -> Result<Vec<u8>, Box<Error>> {
+        let mut encoder = GenericEncoder::new(Vec::new());
+        encoder.value(&payload.construct())?;
+        Ok(encoder.into_inner().into_writer())
+    }
+
+    pub fn intkey_payload_to_transaction(
+        &mut self,
+        payload: IntKeyPayload,
+    ) -> Result<Transaction, Box<Error>> {
+        let mut txn = Transaction::new();
+        let mut txn_header = TransactionHeader::new();
+
+        txn_header.set_family_name(self.addresser.family_name.clone());
+        txn_header.set_family_version(self.addresser.version.clone());
+
+        txn_header.set_nonce(self.rng.gen_ascii_chars().take(50).collect());
+
+        let payload_bytes = self.payload_to_cbor_bytes(&payload)?;
+
+        let mut sha = Sha512::new();
+        sha.input(payload_bytes.as_slice());
+
+        txn_header.set_payload_sha512(sha.result_str());
+
+        txn_header.set_signer_public_key(self.signer.get_public_key()?.as_hex());
+        txn_header.set_batcher_public_key(self.signer.get_public_key()?.as_hex());
+
+        let addresser = IntKeyAddresser::new();
+
+        let addresses = RepeatedField::from_vec(vec![addresser.make_address(payload.name.clone())]);
+
+        txn_header.set_inputs(addresses.clone());
+
+        txn_header.set_outputs(addresses.clone());
+
+        if payload.verb == "inc".to_string() || payload.verb == "dec".to_string() {
+            match self.txn_id_by_name.get(&payload.name) {
+                Some(txn_id) => {
+                    let dependencies = RepeatedField::from_vec(vec![txn_id.clone().to_string()]);
+                    txn_header.set_dependencies(dependencies);
+                }
+                None => {}
+            }
+        }
+        let header_bytes = txn_header.write_to_bytes()?;
+
+        let signature = self.signer.sign(&header_bytes.to_vec())?;
+
+        if payload.verb == "set".to_string() {
+            match self.txn_id_by_name.get(&payload.name) {
```

Comment:
Another case where this and the following match statement can be simplified to if statements